compiler/optimizer/GlobalValuePropagation.hpp

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2000, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#ifndef GLOBALVALUEPROP_INCL
+#define GLOBALVALUEPROP_INCL
+
+#include <stddef.h>                           // for NULL
+#include <stdint.h>                           // for int32_t, int64_t, etc
+#include "compile/Compilation.hpp"            // for Compilation
+#include "cs2/hashtab.h"                      // for HashTable
+#include "env/TRMemory.hpp"                   // for TR_Memory, etc
+#include "env/jittypes.h"                     // for uintptrj_t
+#include "il/DataTypes.hpp"                   // for DataTypes, TR_YesNoMaybe
+#include "il/ILOpCodes.hpp"                   // for ILOpCodes::DIVCHK
+#include "il/Node.hpp"                        // for Node, vcount_t
+#include "infra/Link.hpp"                     // for TR_LinkHead, TR_Link, etc
+#include "infra/List.hpp"                     // for List, TR_ScratchList
+#include "optimizer/Optimization.hpp"         // for Optimization
+#include "optimizer/OptimizationManager.hpp"  // for OptimizationManager
+#include "optimizer/ValuePropagation.hpp"
+
```

Comment:
This question is still valid - I don't know that all these includes need to be here.